pkg/service/webhook/service.go

```diff
@@ -165,7 +166,7 @@ func (s Service) GetSupportedVerbs() GetSupportedVerbsResponse {
 	return GetSupportedVerbsResponse{Verbs: []Verb{Create, Delete}}
 }
 
-func (s Service) PublishWebhook(ctx context.Context, noun Noun, verb Verb, payloadReader io.Reader) {
+func (s Service) PublishWebhook(noun Noun, verb Verb, payloadReader io.Reader) {
 	webhook, err := s.storage.GetWebhook(context.Background(), string(noun), string(verb))
```

Comment:
This should probably use the same context, right?

Reply:
If I use the same context I still get:

ctx, cancel := context.WithTimeout(ctx, duration)
{"level":"warning","msg":"posting payload to http://host.docker.internal:8081/webhook with error: client http client: Post \"http://host.docker.internal:8081/webhook\": context canceled","time":"2023-05-03T18:46:57Z"}

Doing this works:
ctx, cancel := context.WithTimeout(context.background(), duration)

Reply:
My apologies for not being clear. I meant doing the following. So it's the same context throughout the duration of the `PublishWebhook` call. 
```go
func (s Service) PublishWebhook(noun Noun, verb Verb, payloadReader io.Reader) {
  ctx, cancel := context.WithTimeout(context.background(), duration)
  webhook, err := s.storage.GetWebhook(ctx, string(noun), string(verb))
  // some other stuff
  err = s.post(ctx, url, string(postJSONData))
  // .. even more stuff
}
```

Reply:
Good call !